docs/wiki/protocol/data-structures.md

```diff
@@ -0,0 +1,59 @@
+# Data Structures in Ethereum
+
+<br />
+
+## Primer on Merkle Tree
+
+Merkle tree is a hash-based data structure which is very effecient at data integrity and verification. It is a tree based structure where the leaf nodes hold the data values and each non-leaf node is a hash of its child nodes.
+
+![Merkle Tree](../../images/merkle-tree.jpg)
```

Comment:
@varun-doshi great work, but shouldn't the hashes include the child hashes and not only the data? For example, I would replace `H(1, 2, 3, 4)` for `H(H(1, 2), H(3, 4))`.

Reply:
My bad...I'll fix this and push again